monkey/monkey_island/cc/server_setup.py

```diff
@@ -178,8 +178,19 @@ def _connect_to_mongodb(mongo_db_process: Optional[MongoDbProcess]):
         sys.exit(1)
 
 
-def _start_nextjs_server(data_dir: Path):
-    nextjs_process = start_nextjs(data_dir)
+def _start_nextjs_server(config_options: IslandConfigOptions):
+    if is_local_port_in_use(config_options.node_port):
```

Comment:
Not that it's likely or a big deal, but this could result in a TOCTOU bug. Is there a reason we can't just defer to node to figure this out for itself? Aside from the very minor and unlikely TOCTOU issue, it would mean less code for us.

Reply:
I'll test and see what happens. It's definitely less convenient, because the user will have to check node logs to see what happened. If the logs makes sense we can do it, but if not we should leave the port check, as this is a common port.